packages/suite-base/src/App.test.tsx

```diff
@@ -0,0 +1,191 @@
+    expect(screen.getByTestId("global-css")).toBeDefined();
+  });
+
+  it("throw exception when enableGlobalCss is false", () => {
+    setup({ enableGlobalCss: false });
+    expect(() => screen.getByTestId("global-css")).toThrow();
+  });
+
+  it("adds and removes contextmenu event listener on mount/unmount", () => {
+    const addEventListenerSpy = jest.spyOn(document, "addEventListener");
+    const removeEventListenerSpy = jest.spyOn(document, "removeEventListener");
+
+    const { unmount } = setup();
+
+    expect(addEventListenerSpy).toHaveBeenCalledWith("contextmenu", expect.any(Function));
+    expect(removeEventListenerSpy).not.toHaveBeenCalled();
+    unmount();
+    expect(removeEventListenerSpy).toHaveBeenCalledWith("contextmenu", expect.any(Function));
+  });
+
+  it("renders LaunchPreference component when enableLaunchPreferenceScreen is true", () => {
+    setup({ enableLaunchPreferenceScreen: true });
+    expect(screen.getByTestId("launch-preference")).toBeDefined();
+  });
+
+  it("does not render LaunchPreference component when enableLaunchPreferenceScreen is false", () => {
+    setup({ enableLaunchPreferenceScreen: false });
+    expect(screen.queryByTestId("launch-preference")).toBeNull();
+  });
+
+  it("passes deepLinks and onAppBarDoubleClick to Workspace", () => {
+    const mockDeepLinks = ["link1", "link2"];
+    const mockOnAppBarDoubleClick = jest.fn();
+    expect(Workspace).not.toHaveBeenCalled();
+
+    setup({
+      deepLinks: mockDeepLinks,
+      onAppBarDoubleClick: mockOnAppBarDoubleClick,
+    });
+
+    // Ensure Workspace receives the correct props by spying on Workspace
+    expect(Workspace).toHaveBeenCalledWith(
+      {
+        deepLinks: mockDeepLinks,
+        onAppBarDoubleClick: mockOnAppBarDoubleClick,
+      },
+      {},
+    );
+  });
+});
+
+describe("App Component MultiProvider Tests", () => {
+  const expectedProviders = [
+    StudioToastProvider,
+    StudioLogsSettingsProvider,
+    ProblemsContextProvider,
+    CurrentLayoutProvider,
+    UserProfileLocalStorageProvider,
+    LayoutManagerProvider,
+    LayoutStorageContext.Provider,
+  ];
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("verifies that MultiProvider is called with correct providers", () => {
+    setup();
+    expect(screen.getByTestId("multi-provider")).toBeDefined();
+    expect(screen.getByTestId("multi-provider").children).toHaveLength(3);
+  });
+
+  it("verifies that MultiProvider has rendered all providers when its nativeApp", () => {
+    setup({ nativeAppMenu: {} as INativeAppMenu });
+
+    expectedProviders.forEach((provider) => {
+      expect(verifyProviderTypes()).toContain(provider);
+    });
+  });
+
+  it("verifies that MultiProvider has rendered all providers when its nativeWindow", () => {
+    setup({ nativeWindow: {} as INativeWindow });
+
+    expectedProviders.forEach((provider) => {
+      expect(verifyProviderTypes()).toContain(provider);
```

Comment:
here it should verify only if the NativeWindowContext.Provider is rendered, because the nativeWindow is defined.

  if (nativeWindow) {
    providers.push(<NativeWindowContext.Provider value={nativeWindow} />);
  }